classes/class-wc-connect-service-schemas-store.php

```diff
@@ -262,5 +262,20 @@ public function get_predefined_packages_schema() {
 
 			return $predefined_packages;
 		}
+
+		/**
+		 * Returns the WooCommerce Shipping and WooCommerce Tax upgrade banners
+		 *
+		 * @return object|null The banners schema or null if no such id was found
+		 */
+		public function get_wcship_wctax_upgrade_banner() {
+			$service_schemas = $this->get_service_schemas();
+			// check if $service_schemas->features->wcshippingtax_upgrade_banner exists
+			if ( ! is_object( $service_schemas ) || ! property_exists( $service_schemas, 'features' ) || ! property_exists( $service_schemas->features, 'wcshippingtax_upgrade_banner' ) ) {
+				return null;
+			}
+
+			return $service_schemas->features->wcshippingtax_upgrade_banner;
```

Comment:
[Optional - an observation]

Because `property_exists()` returns true even if the value is `null`, there is a chance this line returns `null` causing the `$banner = $schema->get_wcship_wctax_upgrade_banner();` call below to throw "property doesn't exist" error.

If we want to be extra safe, we could use ` ! empty()` or `isset()` here to only return when it isn't `null`.

Reply:
good call, I'll fix it!